lib/galaxy/tools/deps/conda_util.py

```diff
@@ -495,7 +528,14 @@ def build_isolated_environment(
                 export_path
             )
             export_paths.append(export_path)
-        create_args = ["--unknown", "--offline"]
+        create_args = ["--unknown"]
+        # Works in 3.19 and 4.2 - not in 4.0 and 4.3.
+        offline_works = conda_context.conda_version < LooseVersion("4") or \
+            (conda_context.conda_version >= LooseVersion("4.2") and conda_context.conda_version < LooseVersion("4.3"))
+        if offline_works:
+            create_args.extend(["--offline"])
+        else:
+            create_args.extend(["--use-index-cache"])
```

Comment:
I used extend intentionally - it reads better IMO with the rest of the code in that that little block (more uses of the ``extend`` function). Hope this is okay.